pro_tes/middleware/task_distribution/distance.py

```diff
@@ -58,8 +59,10 @@ def task_distribution(input_uri: List) -> List:
     # Calculate distances between all IPs
     try:
         distances = calculate_distance(ips_unique, tes_uri)
-    except (IPDistanceCalculationError, KeyError):
-        return random.task_distribution()
+    except IPDistanceCalculationError as exc:
+        raise IPDistanceCalculationError from exc
+    except KeyError as exc:
+        raise KeyError from exc
```

Comment:
Same as above, just remove the `try`-`except`.

Reply:
Done